README.md

```diff
@@ -415,6 +415,23 @@ Example: `['a', 'b', 'c'] -> 'a, b and c'`
 
 Example: `(' - ', ' or ', ['a', 'b', 'c']) -> 'a - b or c'`
 
+### uniqueString
+
+`array -> string -> string` Returns a function that takes a string and de-duplicates it against an internal cache. Each time this function is called, the resulting deduplicated string is added to the cache. Exposes `cache` and `clear()` properties to access and clear the cache, respectively.
+
+Example: `_.map(uniqueString([]), ['foo', 'foo', 'foo']) -> ['foo', 'foo1', 'foo2']`
+
+### uniqueStringWith
+
+`(cachizer, array) -> string -> string` Allows passing a function (`array -> object`) to override the way `uniqueString`'s initial array is converted into a cache object. Can be curried with a cachizer to create a custom `uniqueString` function.
```

Comment:
`cachizer` might be better named as something indicating that it's about counting occurences 🤔

Reply:
we also probably should mention that it also has cache/clear and maybe give an example

Reply:
It's not necessarily about counting - if there are duplicates in the input array, they only really need to be counted once for `uniqueString` to dedupe properly, so `F.arrayToObject(keyFn, () => 1)` is often just as useful (if not more so) than `_.countBy(keyFn)`. eg:

```js
let dedupeWithCount = uniqueStringWith(
  _.countBy(_.identity), ['foo', 'foo', 'foo', 'bar']
)
let dedupeNoCount = uniqueStringWith(
  F.arrayToObject(_.identity, () => 1), ['foo', 'foo', 'foo', 'bar']
)

_.map(dedupeWithCount, ['foo', 'bar'])  //-> ['foo3', 'bar1']
_.map(dedupeNoCount, ['foo', 'bar'])  //-> ['foo1', 'bar1']
```

These both work fine, but I'd argue that the second one is more intuitive ("what happened to `foo1` and `foo2`?")

So I'm gonna stand by calling it `cachizer` for now - but, yeah, some more details about the function would be a good idea anyway 🙂